superset/connectors/sqla/models.py

```diff
@@ -835,11 +836,9 @@ def get_perm(self) -> str:
             raise DatasetInvalidPermissionEvaluationException()
         return f"[{self.database}].[{self.table_name}](id:{self.id})"
 
-    @property
+    @hybrid_property
     def name(self) -> str:
-        if not self.schema:
-            return self.table_name
-        return "{}.{}".format(self.schema, self.table_name)
+        return self.schema + "." + self.table_name if self.schema else self.table_name
```

Comment:
The logic needed to be translated into a function that SQLAlchemy could understand as we're using this hybrid property as a filter.